### PR TITLE
refactor: naming to RspackPlugin

### PR DESCRIPTION
**Pull request by ahabhgk**
## Summary

naming to XxxRspackPlugin

## Require Documentation?

- [ ] No
- [x] Yes, the corresponding rspack-website PR is \_\_

### CODE REVIEW

**Review by h-a-n-a — DISMISSED** [review]
This diffing is like blowing my mind 🤯. Looks good to me tho.

**Comment by ahabhgk** [comment]
I also change lots of options naming on rust side 😃

File: packages/rspack/src/builtin-plugin/SwcJsMinimizerPlugin.ts [hunk 1]
```diff
@@ -39,24 +39,26 @@ function getMinifyCondition(
...
 	}
 
-	return getMinifyCondition(condition);
+	return getRawSwcJsMinimizerRule(condition);
 }
 
-export const SwcJsMinimizerPlugin = create(
-	BuiltinPluginKind.SwcJsMinimizer,
-	(options?: SwcJsMinimizerPluginOptions): RawMinification => {
+export const SwcJsMinimizerRspackPlugin = create(
```
Comment (9aoy) [inline packages/rspack/src/builtin-plugin/SwcJsMinimizerPlugin.ts:57]: breaking change note.

Is this the final version name? modern has used SwcJsMinimizerPlugin in https://github.com/web-infra-dev/modern.js/pull/4602, lucky we haven't released it yet.

Comment (ahabhgk) [reply packages/rspack/src/builtin-plugin/SwcJsMinimizerPlugin.ts:57]: This is the final version name, there should be no one but modernjs uses SwcJsMinimizerPlugin, cause we didn't ship any docs or release note highlight about it in v0.3.2 intentionally, I consider it is stable to adopt after tomorrow's release (v0.3.3)

Comment (9aoy) [reply packages/rspack/src/builtin-plugin/SwcJsMinimizerPlugin.ts:57]: okk

Comment (h-a-n-a) [reply packages/rspack/src/builtin-plugin/SwcJsMinimizerPlugin.ts:57]: > This is the final version name, there should be no one but modernjs uses SwcJsMinimizerPlugin, cause we didn't ship any docs or release note highlight about it in v0.3.2 intentionally, I consider it is stable to adopt after tomorrow's release (v0.3.3)

Would you add a final naming convention guide for future developers?

Comment (ahabhgk) [reply packages/rspack/src/builtin-plugin/SwcJsMinimizerPlugin.ts:57]: Ofc

**Review by h-a-n-a — APPROVED** [review]

**Comment by 9aoy** [comment]
!canary

**Pull request merged**